.github/workflows/generate-versions.yml

```diff
@@ -0,0 +1,46 @@
+name: Generate RDF Serializations
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - 'ns/*.ttl'
+
+jobs:
+  generate-versions:
+    name: Generate RDF Versions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Install Apache Jena tools (including riot)
+      - name: Install Jena
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y apache2-utils openjdk-11-jre-headless
+          curl -L https://dlcdn.apache.org/jena/binaries/apache-jena-4.8.0.tar.gz \
+            | tar xz --strip-components=1 -C /home/runner
+          echo 'export PATH="/home/runner/bin:$PATH"' >> $GITHUB_ENV
+
+      # Convert all .ttl files to RDF/XML and JSON-LD using riot
+      - name: Convert RDF to multiple formats
+        run: |
+          for file in ns/*.ttl; do
+            base=$(basename "$file" .ttl)
+            # Convert Turtle -> RDF/XML
+            /home/runner/bin/riot --syntax=turtle --output=rdfxml ns/${base}.ttl > ns/${base}.rdf
+            # Convert Turtle -> JSON-LD
+            /home/runner/bin/riot --syntax=turtle --output=jsonld ns/${base}.ttl > ns/${base}.jsonld
+          done
+
+      # Commit and push any changes (RDF/XML, JSON-LD files)
+      - name: Commit generated files
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "actions@github.com"
+          if [[ `git status --porcelain` ]]; then
+            git add ns/*.rdf ns/*.jsonld
+            git commit -m "Generate RDF/XML and JSON-LD for updated vocabularies"
+            git push
+
```

Comment:
@domel My mantra has always been "do not commit generated files".
I think we should deploy them on github pages, but not necessarily commit them back to the repo. Some people (including our future selves) may not understand/remember that these files are generated, and feel compelled to updated them manually. Or try to fix some syntactic aspects of the JSON or RDF/XML (e.g. indentation) and be frustrated that their changes are automatically reverted...

Reply:
In some cases, it is worth committing generated files (a/k/a artefacts or artifacts), to make the results of "source" changes more easily visible, but in such case, it is best to direct them to a different folder within the repo, so it is clear that they *are* artifacts.

Reply:
```suggestion
```